### DISCUSSION TITLE
Roller compression analysis: null pressure output

### DISCUSSION THREAD

**Discussion by yvanblanchard** [adtzlr/felupe#859] [Q&A] [answered]
Hello,

I used a previous script using Felupe for computing radial deformation and pressure contact of a rubber roller on flat shape.

This model integrates a Mooney-Rivlin model for the rubber, and isotropic material for the roller coating (outer skin).
The computations converge, but the output pressure seems to be null (and output plot of pressure versus contact width is not generated).

Could you please help ?

here is the python script:
[RollerCompaction-Wang-UnivBristol.py.txt](https://github.com/user-attachments/files/17110936/RollerCompaction-Wang-UnivBristol.py.txt)

**Comment by yvanblanchard** [comment]
Remark: it is strange, but this script does not work if launched as a notebook in VS Code (does not converge):

![image](https://github.com/user-attachments/assets/79340dee-2581-4bc5-ba9b-76e0410d713b)

**Comment by adtzlr** [comment, marked as answer]
Hi @yvanblanchard,

you have to replace

```python
boundary = fem.RegionQuadBoundary(container.meshes[0], mask=outer, ensure_3d=True)
```

with

```python
boundary = fem.RegionQuadBoundary(container.meshes[1], mask=outer, ensure_3d=True)
```

or

```python
boundary = fem.RegionQuadBoundary(mesh, mask=outer, ensure_3d=True)
```

because the first sub-mesh `container.meshes[0]` does not contain cells which are connected to points on the outer radius.

**Comment by yvanblanchard** [comment]
Thank you @adtzlr , it works fine
And I also noticed that it works also fine in a notebook (in VS Code IDE), but need to regroup/merge some notebook cells (don't know why, maybe a VSCode bug).
